tests/integration/kubernetes/confidential_kbs.sh

```diff
@@ -157,17 +157,7 @@ kbs_set_resource_from_file() {
 
 	kbs_pod=$(kubectl -n $KBS_NS get pods -o NAME)
 	kbs_repo_path="/opt/confidential-containers/kbs/repository"
-	for i in {1..3}
-	do
-		sleep 5
-		kubectl -n $KBS_NS exec -it "$kbs_pod" -- \
-			bash -c "ls $kbs_repo_path/$path" && \
-			return
-
-		>&2 echo "WARN: resource '$path' was not created yet, retrying ..."
-	done
-	>&2 echo "ERROR: resource '$path' could not be created yet"
-	return 1
+	kubectl -n $KBS_NS exec -it "$kbs_pod" -- bash -c "for i in {1..30}; do [ -e '$kbs_repo_path/$path' ] && exit 0; sleep 0.5; done; exit -1"
```

Comment:
What I had in mind was something like

```bash
if ! kubectl -n $KBS_NS exec -it "$kbs_pod" -- bash -c "for i in {1..30}; do [ -e '$kbs_repo_path/$path' ] && exit 0; sleep 0.5; done; exit -1"; then
    echo "ERROR: resource '$path' not created in 15s"
    kubectl -n $KBS_NS exec -it "$kbs_pod" -- bash -c "find $kbs_repo_path"
fi
```
which should provide similar output (with the possibility for the path to be created between the last check and the first error-handling check.

Reply:
Squashed your suggestion to the first patch, dropped the second patch, and added a few more patches to help.